EIPS/eip-3085.md

```diff
@@ -37,13 +37,9 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
 ### `wallet_addEthereumChain`
 
 The method accepts a single object parameter, with a `chainId` and some chain metadata.
-The method returns `null` if the chain was added to the wallet, and an error otherwise.
+The method returns `null` if the requested chain was added to the wallet as a result of the method call, and an error otherwise.
```

Comment:
I feel like this should be a success if the chain was previously added with the same parameters.  As a caller, what I care about is that the chain is present.  I don't really care whether this was the first time it was added or if it was added in a previous call.

There are some security issues with a malicious dapp adding chain 123 and then an honest dapp adding chain 123 but with otherwise different configuration, so there may be value in returning an error if the two chains are sufficiently different.  On the other hand, the user may be using their own custom block explorer, or their own JSON-RPC endpoint which would cause the chain added to not exact match.

In other words, this is all complicated, but I feel like an error is not the correct result in the case of a duplicate chain.

Reply:
To put it another way: This error is almost certainly going to be expressed as an exception in the vast majority of libraries, and exceptions should not be used for flow control, and this "error" will occur every time after the first time a user interacts with a particular chain.

Reply:
One potential solution would be to provide a mechanism for first asking if a particular chain is available.  This would allow the app developer to make this request only when it is actually needed, rather than requiring the developer to use exception handling for flow control.